internal/vpn/wireguard.go

```diff
@@ -22,7 +21,19 @@ func setupWireguard(ctx context.Context, netlinker NetLinker,
 		return nil, "", fmt.Errorf("finding a VPN server: %w", err)
 	}
 
-	wireguardSettings := utils.BuildWireguardSettings(connection, settings.Wireguard, ipv6Supported)
+	var wireguardSettings wireguard.Settings
+
+	// if provider has WireguardConfigurator interface, use it
+	wireguardConfigurator, ok := providerConf.(provider.WireguardConfigurator)
+	if ok {
+		// Use already defined variables
+		wireguardSettings, err = wireguardConfigurator.GetWireguardConnection(ctx, connection, settings.Wireguard, ipv6Supported)
+		if err != nil {
+			return nil, "", fmt.Errorf("getting Wireguard connection: %w", err)
+		}
+	} else {
+		wireguardSettings = utils.BuildWireguardSettings(connection, settings.Wireguard, ipv6Supported)
+	}
```

Comment:
It would be better to inject `ctx` + `providerConf` into `utils.BuildWireguardSettings` and modify `buildWireguardSettings` to check if providerConf has the `GetWireguardConfig` method.

Also, if you do this with a type assertion as you do now, it will pass for all providers since each have the `GetWireguardConfig` method from the `NoWireguardXXX` interface; I would suggest to simply remove the `NoWireguardXXX` interface you defined, and that would nicely solve it.